### PR TITLE
[WIP] Direct link to file/line in commits and pull requests

### LINKED ISSUE #413: Direct link to line in commit doesn't work correctly
Instead of opening code view for the selected file with the specified line highlighted it is only opening commit screen and doesn't go further than that. 

Example: https://github.com/slapperwan/gh4a/commit/c021f4585244ba4a4cfdb90203aec1034818a47c#diff-1e105dc4c4e8c142976247ec34a5f610R275

### PR DESCRIPTION
**Pull request by Tunous**
Added it both for the links to files in commits and pull requests.

@maniac103 Need your help with line highlighting. I don't see it being implemented for file diffs and am not really sure how to do it correctly.

Closes #413.

### CODE REVIEW

**Comment by Tunous** [comment]
A bug I found that might be ok to fix in this pull request:

Navigating up from pull request file diff goes to pull requests list instead of pull request.

**Comment by maniac103** [comment]
@Tunous I hope you don't mind me updating this branch. Please let me know what you think of this solution.

File: src/com/gh4a/activities/DiffViewerActivity.java [hunk 1]
```diff
@@ -192,15 +206,36 @@ protected String generateHtml(String cssTheme, boolean addTitleHeader) {
...
             } else if (line.startsWith("+")) {
+                ++rightDiffPosition;
                 cssClass = "add";
             } else if (line.startsWith("-")) {
+                ++leftDiffPosition;
                 cssClass = "remove";
+            } else {
+                ++leftDiffPosition;
+                ++rightDiffPosition;
+            }
```
Comment (maniac103) [inline src/com/gh4a/activities/DiffViewerActivity.java]: The logic I added here should also allow fixing #409.

**Comment by Tunous** [comment]
It's ok but that 'duplicated fetch logic' was done on purpose. With it users are redirected directly to the diff activity which has these advantages:

- They don't see commit activity while the commit is still loading
- Instead they see proper loading dialog
- Pressing back redirects them back to where they were instead of commit activity (they can go to commit using up arrow)

IMO this behavior is much better than opening 2 activities. It results in better back stack behavior and less screen jumps.

**Comment by maniac103** [comment]
I think I do the same thing when going to a specific release though ... adapting that means adding more duplicated logic to BrowseFilter, which is something I really want to avoid to get into (because there's not only the code duplication, but also a high chance of subtle behavior differences AKA subtle bugs). I also tend to think of the URL fragments as 'initial navigation hints', not as something that completely should change the activity a using is seeing ... so I guess we have different views on this topic. :-/

**Comment by maniac103** [comment]
After some more thinking I think you're right though. I'm just wondering whether showing only the diff viewer provides enough context, as its title bar info is rather sparse.
Feel free to revert back to your version of the code, otherwise I'll do it tomorrow. I'd like to keep the pre-parsing of the fragment before starting a background task, though.

**Comment by Tunous** [comment]
https://github.com/Tunous/TestingRepo/releases/tag/0.0.1

**Comment by Tunous** [comment]
I think that these load tasks together with the progress dialog and a new base class (for the common dialog showing/error handling code) should be moved outside of `BrowseFilter`. Perhaps to their own package each as a separate file.

**Comment by Tunous** [comment]
> I'm just wondering whether showing only the diff viewer provides enough context, as its title bar info is rather sparse.

I don't think that's a problem. Users can just press on the up arrow and see the matching commit with more information.

**Comment by maniac103** [comment]
Why should those tasks be moved out? AFAICT all of them are specific to loading stuff from URLs - at least the RefPathDisambiguationTask surely is.

**Comment by Tunous** [comment]
No real reason for that. Just a simple suggestion since they seem to be more for direct launching of different screens than parsing URLs. (Except for the RefPathDisambiguationTask)

**Comment by maniac103** [comment]
👍 from my side now. @Tunous are you also fine with it?

**Comment by Tunous** [comment]
Yep :+1:

**Comment by maniac103** [comment]
Shipit then, thanks for your patience with my back-and-forth thoughts :)

**Pull request merged**